### PR TITLE
Let plugins control modal layout, scrolling and clipboard actions

### PR DESCRIPTION
**Pull request by boudra**
### Linked issue

Refs #3930. Its reported ordinary-surface crash was not reproduced; this PR does not claim to resolve that report.

### Type of change

- [x] Bug fix
- [x] New feature
- [x] Enhancement
- [x] Docs

### Reasoning

Plugin authors can paint the entire modal body, override its padding, and choose a single vertical scroll owner. Custom lists scroll at the initial native sheet height, and SDK inputs keep keyboard focus registered with the sheet. A supported `copyText` action reports success or failure on the client clipboard.

The host previously forced its scrolling body and decorative padding. Native custom lists also had their offsets reset below the largest sheet detent. The shared sheet now separates body styling, decorative spacing and safe-area clearance; author-owned lists take body gestures while the handle retains resize/dismiss behavior. Paseo's model browser and diagnostic views use the same presentation-aware scrolling boundary.

### Goals

- Allow default, custom and zero body padding across compact and wide layouts while retaining safe-area clearance once.
- Support ScrollView, FlatList, horizontal tabs, clipboard actions and keyboard-aware inputs through the plugin SDK.
- Document composition and provide a runnable example exercised by the browser regression.

### Non-goals

- Automatically rewrite third-party plugins or identify which interaction caused the original author report.
- Change protocol compatibility or add clipboard-reading permission.
- Supersede #3972: its additional Android popover tap repair remains outside this change.

### QA

Tested unreleased 0.8 code with the Android 0.7.2 development binary, API 35, RN 0.81.5, Gorhom 5.2.14. Every runtime test used an isolated daemon.

- Android: swiped ScrollView/FlatList content; imperative jump reached row 100 at the initial 65% sheet height and stayed there after handle expansion. Before the gesture ownership change, Gorhom reset the list offset below its top detent.
- Android clipboard: Copy text followed by long-press OS Paste inserted `Copied from Paseo`. Selectable text exposed OS Copy/Select all. Real Gboard covered the plain RN input; the SDK input raised the sheet and remained visible.
- Android compact and wide layouts: custom padding and bounded lists; same example in a workspace panel. First-party project footer remained above the keyboard and diagnostics scrolled. Model picker opened, but populated model selection was not exercised in the isolated host.
- Chromium compact and wide: default 24px, custom 8px/40px and zero padding; wheel/ref scrolling; horizontal tabs; copy/paste; explicit clipboard-denied error.
- Local checks: workspace build, root typecheck/lint/format, 72 targeted unit tests across eight files, and four browser regressions (host UI, modal bodies, plugin management and workspace panels). No full local suite.

Screenshots and reproduction steps: [modal UI example](https://github.com/getpaseo/paseo/tree/diagnose/plugin-mobile-scroll/plugin-examples/modal-ui).

Risk surface: shared native sheet sizing, safe-area spacing and body gesture ownership also affect first-party sheets. iOS, real iPad and Electron-specific clipboard policy were not tested on this Linux host. Existing plugins must adopt the new SDK components and ownership pattern.

### Checklist

- [x] One focused change
- [x] `npm run typecheck` passes
- [x] `npm run lint` passes
- [x] `npm run format` passes
- [x] QA evidence
- [x] Tests added or updated where it made sense

### CODE REVIEW

File: packages/app/src/components/ui/scroll-view.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { isNative } from "@/constants/platform";
+import { useIsInsideBottomSheet } from "./bottom-sheet-scope";
+
+// Both libraries forward to the RN FlatList instance. Their published ref types
+// disagree about getScrollResponder (Gorhom) and include the component function
+// in the instance intersection (RNGH); keep that type discrepancy at this boundary.
+const SheetFlatList = BottomSheetFlatList as unknown as typeof NativeFlatList;
+const HorizontalFlatList = GestureFlatList as unknown as <Item>(
+  props: FlatListProps<Item> & NativeViewGestureHandlerProps & { ref?: Ref<NativeFlatList<Item>> },
+) => React.ReactElement;
```
[the remark this answers is not in the document]
Comment (boudra) [reply packages/app/src/components/ui/scroll-view.tsx:23]: These are intentional corrections at the third-party boundary, rather than conversions between different runtime components. I checked the installed sources: Gorhom's `BottomSheetFlatList.tsx` wraps RN FlatList and `createBottomSheetScrollableComponent.tsx` forwards `scrollableRef.current` through `useImperativeHandle`; RNGH's `GestureComponents.tsx` passes the same ref directly to `<RNFlatList>`. Gorhom's declaration differs on `getScrollResponder`, while RNGH declares the instance as a component-function/instance intersection. Those declaration mismatches are documented next to the adapters.

The other two assertions restore the item generic erased by React's `forwardRef`; they do not convert the underlying ref. The native and browser QA exercised the forwarded `scrollToEnd` ref, including Android at the initial sheet snap. Replacing this with a narrower public ref API or runtime component checks would change the supported contract to accommodate incorrect upstream declarations. Keeping the isolated adapters for this PR; this finding does not identify a runtime incompatibility to repair.

File: packages/app/e2e/browser/plugin-modal-body.spec.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      await closeExample(page);
+      await cdp.detach();
+    });
+  } finally {
+    await client.removePlugin("modal-ui-example").catch(() => undefined);
+    await client
+      .patchDaemonConfig({ pluginsEnabled: previous.config.pluginsEnabled ?? false })
+      .catch(() => undefined);
+    await client.close().catch(() => undefined);
+  }
```
[the remark this answers is not in the document]
Comment (boudra) [reply packages/app/e2e/browser/plugin-modal-body.spec.ts]: Moved daemon installation/configuration and cleanup into a fixture, and viewport navigation and denied-permission mechanics into named helpers. The test body now declares five lines of user intent. All existing assertions remain; the real browser regression passes (22.1s), as do root typecheck, lint and formatting. Fixed in 8311a0427.
